### DISCUSSION TITLE
Installing homebrew on Mac M1 silicon vs Rosetta 2

### DISCUSSION THREAD

**Discussion by bobettebob** [Homebrew/discussions#2845] [Getting started] [answered]
Do I need to install Rosetta2 before installing Homebrew and other package if I plan to compiling audio-video open source app that normally work with Mac intel chip? It seem to have trouble generating config.INI at first running.
When I 'brew config',  Rosetta 2: false
Thank you!

**Comment by gromgit** [comment]
Sounds like you'd need to install Rosetta, but you might first want to check if it hasn't already been installed:
```
echo Rosetta is $(arch -x86_64 /usr/bin/true || echo not) installed
```
If the above command says it's not installed:
```
softwareupdate --install-rosetta --agree-to-license
```
should do it for you. You'd also need to install Homebrew-for-Intel:
```
arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```
then open a separate shell and:
```
eval "$(/usr/local/bin/brew shellenv)"
```
then install all the Intel formulae you need, and do all your Intel work there.

I think that's about all you need to do.

**Comment by bobettebob** [comment, marked as answer]
Do I need to uninstall my homebrew  arm(silicon chip) and all my others packages installed through homebrew arm (wget, command line tool for xcode, free pascal, lazarus) before installing homebrew for intel?
Thank you!

**Reply by gromgit** [reply]
No, both Intel (`/usr/local`) and Apple Silicon (`/opt/homebrew`) Homebrew hierarchies can coexist just fine.

However, if you have both hierarchies installed, you almost certainly want to:
* avoid installing the same formula in both hierarchies unless absolutely necessary
* be explicit about pathnames (e.g. `/usr/local/bin/gcc` instead of just `gcc`)
